app/components/SignUp/index.jsx

```diff
@@ -182,6 +194,16 @@ export default class SignUp extends React.Component {
       email: e.target.value,
     };
 
+    if (!/\S+@\S+\.\S+/.test(e.target.value)) {
```

Comment:
Of course it would make sense to move validation logic into a function, as it's used in multiple places, but for now it's not critical, so I don't mind to merge it as is.